sdk/program/src/stake/tools.rs

```diff
@@ -0,0 +1,29 @@
+//! Utility functions
+
+/// Helper function for programs to call [`GetMinimumDelegation`] and then fetch the return data
+///
+/// This fn handles performing the CPI to call the [`GetMinimumDelegation`] function, and then
+/// calls [`get_minimum_delegation_return_data()`] to fetch the return data.
+///
+/// [`GetMinimumDelegation`]: super::instruction::StakeInstruction::GetMinimumDelegation
+pub fn get_minimum_delegation() -> Option<u64> {
+    let instruction = super::instruction::get_minimum_delegation();
+    crate::program::invoke(&instruction, &[]).ok()?;
+    get_minimum_delegation_return_data()
+}
+
+/// Helper function for programs to get the actual data after calling [`GetMinimumDelegation`]
+///
+/// This fn handles calling [`get_return_data()`], ensures the result is from the correct
+/// program, and returns the correct type.  Returns `None` otherwise.
+///
+/// [`GetMinimumDelegation`]: super::instruction::StakeInstruction::GetMinimumDelegation
+/// [`get_return_data()`]: crate::program::get_return_data
+pub fn get_minimum_delegation_return_data() -> Option<u64> {
```

Comment:
I don't think we need to expose this function publicly because I can't think of a use-case where a program wouldn't just call `stake::tools::get_minimum_delegation`.

Reply:
Gotcha. Yeah, I'll defer to you here. I see that `invoke()` returns a `Result`, so I wasn't sure if any programs would want to call `GetMinimumDelegation` directly, do something with the result, then call this helper fn. Would that ever happen? If no, then I agree there's no need for this to be public.